Update scala to 2.13.3 and silencer to 1.7.0

raboof opened the pull request.

raboof committed:
Update scala to 2.13.2 and silencer to 1.7.0

patriknw commented on project/Dependencies.scala:
```diff
@@ -26,7 +26,7 @@ object Dependencies {
 
   val scala212Version = "2.12.11"
-  val scala213Version = "2.13.1"
+  val scala213Version = "2.13.2"
```
also .travis.yml

patriknw replied:
done

raboof committed:
Also travis

patriknw commented:
I compared the Artery TCP MaxThroughputSpec with 2.12, 2.13.1 and 2.13.2. Should be some ByteString (Vector) usage there. Results showed small 5% improvement between 2.13.1 and 2.13.2, but maybe within error margin because I only ran it once.

johanandren commented:
I noticed as well, by chance, the other day that 2.13.2 has become more particular in what it accepts and fails the build.

patriknw commented:
@raboof anything holding us back from completing and merging this?

raboof commented:
> anything holding us back from completing

No, I just lost track of this one ;)

raboof committed:
Fix various warnings

Perhaps the lookup() change was a bit too much?

2 warnings that need a closer look remain

raboof committed:
Fix more warnings

raboof committed:
2 duplicate headers in multi-jvm tests

patriknw commented on akka-actor/src/main/scala-2.13/akka/util/ByteString.scala:
```diff
@@ -1331,7 +1331,7 @@ final class ByteStringBuilder extends Builder[Byte, ByteString] {
   }
 
-  def result: ByteString =
+  def result(): ByteString =
```
ok, that should be source compatible, although it might introduce warning

patriknw commented on akka-actor/src/main/scala/akka/actor/Extension.scala:
```diff
@@ -132,5 +132,5 @@ trait ExtensionIdProvider {
    * Returns the canonical ExtensionId for this Extension
    */
-  def lookup(): ExtensionId[_ <: Extension]
+  def lookup: ExtensionId[_ <: Extension]
```
is this source compatible?

raboof replied:
Looks like it, with `-Dakka.no.discipline=true` subclasses with parens are still allowed (checked both 2.12 and 2.13)

patriknw replied:
ok, thanks

patriknw reviewed (commented):
looking good, one question about source compatibility...

patriknw reviewed (approved):
LGTM

raboof committed:
2.13.3

raboof committed:
scalafmt

raboof commented:
PLS BUILD

ennru commented on .travis.yml:
```diff
@@ -4,7 +4,7 @@ sudo: false
 scala:
   - "2.12.11"
-  - "2.13.1"
+  - "2.13.2"
```
!

raboof replied:
```suggestion
  - "2.13.3"
```

raboof committed:
Also update travis to 2.13.3

raboof commented:
This `AsyncDnsResolverSpec` seems pretty consistent on this branch and I haven't seen it on others, so might actually be a problem with this PR.

raboof committed:
Check both the name and the IP address

raboof committed:
Missing parens

patriknw commented:
PLS BUILD

patriknw commented:
This can be merged, but having an extra validation run first
